src/logger.ts

```diff
@@ -149,36 +155,40 @@ export function setupLogEvents(): void {
 }
 
 // this function save recorded transaction to sqlite with its tx type
-export async function txStatusSaver(_txs: DetailedTxStatus[]): Promise<void> {
-  const txs = _txs
-
-  const bulkInsertTxs = true
-  if (bulkInsertTxs) {
-    if (txs.length === 0) return
-    const prepareBulkInsertSQL = (txs: DetailedTxStatus[]): string => {
-      // items order {txHash, injected, accepted, reason, type, to, from, ip, timestamp}
-      // eslint-disable-next-line prefer-const
-      let { txHash, injected, accepted, reason, type, to, from, ip, timestamp, nodeUrl } = txs[0]
-      // nodeUrl = nodeUrl ? nodeUrl : new URL(nodeUrl as string).hostname
-
-      let placeholders = `NULL, '${txHash}', '${type}', '${to}', '${from}', '${injected}', '${accepted}', '${reason}', '${ip}', '${timestamp}', '${nodeUrl}'`
-      let sql = 'INSERT OR REPLACE INTO transactions VALUES (' + placeholders + ')'
-
-      for (let i = 1; i < txs.length; i++) {
-        // eslint-disable-next-line prefer-const
-        let { txHash, injected, accepted, reason, type, to, from, ip, timestamp, nodeUrl } = txs[i] // eslint-disable-line security/detect-object-injection
-        // nodeUrl = nodeUrl ? nodeUrl : new URL(nodeUrl as string).hostname
-        placeholders = `NULL, '${txHash}', '${type}', '${to}', '${from}', '${injected}', '${accepted}', '${reason}', '${ip}', '${timestamp}', '${nodeUrl}'`
-        sql = sql + `, (${placeholders})`
+export async function txStatusSaver(txs: DetailedTxStatus[]): Promise<void> {
+  if (txs.length === 0) return
+
+  try {
```

Comment:
same as above.  additional error handinling and finalize statment

```suggestion
  try {
  const stmt = db.prepare(`
    INSERT OR REPLACE INTO transactions 
    ("hash", "type", "to", "from", "injected", "accepted", "reason", "ip", "timestamp", "nodeUrl") 
    VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
  `);

  const insertMany = db.transaction((items) => {
    try {
      for (const item of items) {
        stmt.run(
          item.txHash,
          item.type,
          item.to,
          item.from,
          item.injected ? 1 : 0,
          // if accepted is a boolean, convert it to the corresponding TxStatusCode
          // otherwise, use the value as is
          typeof item.accepted === 'boolean' 
            ? (item.accepted ? TxStatusCode.SUCCESS : TxStatusCode.BAD_TX) 
            : item.accepted,
          item.reason || null,
          item.ip || null,
          item.timestamp,
          item.nodeUrl || null
        );
      }
    } catch (error) {
      console.error('Error during transaction:', error);
      throw error
    } finally {
      stmt.finalize()
    }
  });

  insertMany(txs);
} catch (e) {
  console.error('Error inserting transactions:', e);
}

```

Reply:
no use of finalize in better-sqllite3.  ignore